leptos_dom/src/html.rs

```diff
@@ -276,6 +276,12 @@ impl ElementDescriptor for Custom {
 cfg_if! {
   if #[cfg(all(target_arch = "wasm32", feature = "web"))] {
     /// Represents an HTML element.
+    /// Beginner's tip: 
+    /// HtmlElement<El> implements Deref where El implements Deref.
+    /// When El has a corresponding web_sys::HtmlElement -> El will implement Deref for it.
+    /// For instance leptos::HtmlElement<Div> impl Deref for web_sys::HtmlDivElement
+    /// Because of [Deref Coercion](https://doc.rust-lang.org/std/ops/trait.Deref.html#deref-coercion) you can call applicable web_sys::HtmlElement methods on HtmlElement<El> as if it were that type.
```

Comment:
Maybe add something about needing to occasionally manually deref. Feel free to copy the below.

If both `HtmlElement` and one of its derefs have a method with the same name, the dot syntax will call the method on the inherent impl (i.e. `HtmlElement`). You may need to manually deref to access other methods, for example, [`(*el).style()`](https://docs.rs/web-sys/latest/web_sys/struct.HtmlElement.html#method.style) to get the [`CssStyleDeclaration`](https://docs.rs/web-sys/latest/web_sys/struct.CssStyleDeclaration.html) instead of calling [`leptos::HtmlElement::style`].